File: Python/stack_queues/queue.py
# implementing a queue


class Queue(object):
    def __init__(self):
        self.queue = []

    def enqueue(self, value):
        self.queue.append(value)
        return

    def dequeue(self):
        return self.queue.pop(0)

    def print_queue(self):
        print self.queue
        return


if __name__ == "__main__":
    queue = Queue()
    queue.enqueue(1)
    queue.enqueue(2)
    queue.enqueue(3)
    queue.enqueue(4)
    queue.print_queue()
    print queue.dequeue()
    queue.print_queue()

File: PHP/data_structures/arrays/queue.php
<?php

/**
 * Implement a queue data structure with arrays.
 */
class Queue {
    private $arr;

    public function __construct() {
        $this->arr = [];
    }

    /**
     * Queue an element to the end of the queue.
     * Time: O(1)
     * @param  [type] $data [description]
     * @return [type]       [description]
     */
    public function enqueue($data) {
        $this->arr[] = $data;
    }

    /**
     * Remove an element from the start of the queue.
     * Time: O(1)
     * @return [type] [description]
     */
    public function dequeue() {
        if ($this->isEmpty()) {
            echo "Queue is empty." . PHP_EOL;
            return;
        }
        return array_shift($this->arr);
    }

    public function isEmpty() {
        return empty($this->arr);
    }

    public function printQueue() {
        echo implode(" ", $this->arr) . PHP_EOL;
    }
}

$queue = new Queue();
$queue->enqueue(1);
$queue->enqueue(2);
$queue->enqueue(3);
$queue->enqueue(4);
$queue->enqueue(5);

$queue->dequeue();
$queue->printQueue();
$queue->dequeue();
$queue->printQueue();
$queue->dequeue();
$queue->printQueue();
$queue->dequeue();
$queue->printQueue();
$queue->dequeue();
$queue->printQueue();
$queue->dequeue();
